packages/scm/src/browser/scm-widget.tsx

```diff
@@ -84,6 +86,14 @@ export class ScmWidget extends BaseWidget implements StatefulWidget {
                 this.updateViewMode(e.newValue);
             }
         }));
+        this.toDispose.push(this.shell.onDidChangeCurrentWidget(({ newValue }) => {
+            if (newValue instanceof EditorWidget) {
+                const uri = newValue.getResourceUri();
+                if (uri) {
+                    this.resourceWidget.selectNodeByUri(uri);
+                }
+            }
```

Comment:
Suggestion: I believe we can simplify this (and make it less specific to `EditorWidget`).

```suggestion
            const uri = NavigatableWidget.getUri(newValue || undefined);
            if (uri) {
                this.resourceWidget.selectNodeByUri(uri);
            }
```